packages/fax/src/rest/v3/emails/emails-api.ts

```diff
@@ -0,0 +1,224 @@
+import { Email, EmailRequest, ServicePhoneNumber, UpdateEmailRequest } from '../../../models';
+import {
+  ApiListPromise,
+  buildPageResultPromise,
+  createIteratorMethodsForPagination,
+  PaginatedApiProperties,
+  PaginationEnum,
+  RequestBody,
+  SinchClientParameters,
+} from '@sinch/sdk-client';
+import { FaxDomainApi } from '../fax-domain-api';
+
+export interface AddEmailToNumbersRequestData {
+  /**  */
+  'emailRequestBody': EmailRequest;
+}
+export interface DeleteEmailRequestData {
+  /** The email you want to delete. */
+  'email': string;
+}
+export interface ListEmailsForProjectRequestData {
+  /** Number of items to return on each page. */
+  'pageSize'?: number;
+  /** Optional. The page to fetch. If not specified, the first page will be returned. */
+  'page'?: string;
+}
+export interface ListNumbersByEmailRequestData {
+  /** The email you want to get numbers for. */
+  'email': string;
+  /** Number of items to return on each page. */
+  'pageSize'?: number;
+  /** Optional. The page to fetch. If not specified, the first page will be returned. */
+  'page'?: string;
+}
+export interface UpdateEmailRequestData {
+  /** The email you want to work with. */
+  'email': string;
+  /**  */
+  'updateEmailRequestBody'?: UpdateEmailRequest;
+}
+
+export class EmailsApi extends FaxDomainApi {
+
+  /**
+   * Initialize your interface
+   *
+   * @param {SinchClientParameters} sinchClientParameters - The parameters used to initialize the API Client.
+   */
+  constructor(sinchClientParameters: SinchClientParameters) {
+    super(sinchClientParameters, 'EmailsApi');
+  }
+
+  /**
+   * Add an email
+   * Add an email to be used for sending and receiving faxes.
+   * @param { AddEmailToNumbersRequestData } data - The data to provide to the API call.
+   */
+  public async addToNumbers(data: AddEmailToNumbersRequestData): Promise<Email> {
+    this.client = this.getSinchClient();
+    const getParams = this.client.extractQueryParams<AddEmailToNumbersRequestData>(data, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json',
+    };
```

Comment:
Those headers seem to be the same for all `EmailsApi` operation. Maybe it will be better to not duplicate the code and use class variable for that?

Reply:
I don't think it would be better to factorize in the code what can be (and actually is) defined individually at operation level in the specification. If the specification is updated later on and changes a single operation definition, the impacts on this generated code will be much bigger to apprehend than a direct regeneration.
What is bothering you exactly in this file? Is it the repetition of the same string? Is it the object definition itself?

Reply:
yeah; just the repetition thing. Personally I like more concise source files.